File: Satisfiability/3SAT.1.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <bitset>
using namespace std;

int satisfiable(int n, int m, vector<vector<int>> F, vector<int>& S) {
    for (int v = 0; v < pow(2, n); v++) {
        // cout << "v = " << v << endl;
        for (int i = 0; i < m; i++) {
            int t = v;
            for (int j = 0; j < n; j++) {
                if (F[i][j] == 0)
                    t ^= (1 << (n - j - 1));
            }
            // cout << "\t" << bitset<3>(t) << endl;
            if (t == 0)
                return v;
        }
    }
    return -1;
}

int main() {
    int n, m; 
    cin >> n >> m;
    vector<vector<int>> F(m, vector<int>(n));
    for (int i = 0; i < m; i++)
        for (int j = 0; j < n; j++)
            cin >> F[i][j];
    vector<int> S;
    int result = satisfiable(n, m, F, S);
    if (result < 0)
        cout << "No" << endl;
    else
        cout << bitset<3>(result) << endl;
}

File: Satisfiability/3SAT.1.py
def satisfiable(n, m, b):
    for v in range(2**n):
        for i in range(m):
            t = v
            for j in range(n):
                if b[i][j] == 0:
                    t ^= (1 << (n - j - 1))
            if t == 0:
                return True, v
    return False, None

n, m = map(int, input().split())
b = []
for i in range(m):
    b.append(list(map(int, input().split())))
result, solution = satisfiable(n, m, b)
print("Satisfiable:", result)
if result:
    for i in range(n):
        print(solution & 1, end = " ")
        solution >>= 1
        
   